src/group/mls_group/new_from_welcome.rs

```diff
@@ -32,13 +31,27 @@ impl MlsGroup {
             return Err(e);
         }
 
-        // Compute keys to decrypt GroupInfo
-        let (mut group_info, intermediate_secret, path_secret_option) = Self::decrypt_group_info(
-            &ciphersuite,
-            &egs,
-            key_package_bundle.private_key(),
-            welcome.encrypted_group_info(),
+        let group_secrets_bytes = ciphersuite.hpke_open(
+            &egs.encrypted_group_secrets,
+            &key_package_bundle.private_key(),
+            &[],
+            &[],
         )?;
+        let group_secrets = GroupSecrets::decode_detached(&group_secrets_bytes)?;
+        let joiner_secret = group_secrets.joiner_secret;
+        // TODO #141: Implement PSK
+        let mut key_schedule = KeySchedule::init(ciphersuite, joiner_secret, None);
+        let (welcome_key, welcome_nonce) = key_schedule
+            .welcome()?
+            .derive_welcome_key_nonce(ciphersuite);
+
+        let group_info_bytes =
+            match welcome_key.aead_open(welcome.encrypted_group_info(), &[], &welcome_nonce) {
```

Comment:
I think `ok_or(...)` would be a bit nicer here.

Reply:
Sure, will do (`map_err` it is actually). But same comment as above. This code is only moved around and I don't want to touch it too much in this PR.

Reply:
I thought you use `map_err` if you want to use the existing error to compute the new one. `ok_or` just ignores the existing error by default and just outputs the one you specify.